pkg/cloudprovider/suite_test.go

```diff
@@ -314,6 +323,104 @@ var _ = Describe("CloudProvider", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(isDrifted).To(BeTrue())
 		})
+		It("should return drifted if the AWSNodeTemplate securitygroup do not match the machine", func() {
```

Comment:
A sanity check for the empty case. It's not possible to have an empty set of security groups, right?

Reply:
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-security-groups.html#:~:text=When%20you%20launch%20an%20instance%20in%20a%20VPC%2C%20you%20must%20specify%20a%20security%20group%20that%27s%20created%20for%20that%20VPC. Looks like an instance has to have security groups, and the node template can only be empty if the ANT controller hasn't updated the status due to race or an error.

Reply:
In the empty case, drift will just error out. We will not drift